opnode/node/driver.go

```diff
@@ -0,0 +1,105 @@
+package node
+
+import (
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/trie"
+	"github.com/holiman/uint256"
+)
+
+type InputItem struct {
+	Block    *types.Block
+	Receipts []*types.Receipt
+}
+
+func (item *InputItem) Epoch() uint64 {
+	return item.Block.NumberU64()
+}
+
+// sanity check that the receipts are consistent with the block data
+func (item *InputItem) CheckReceipts() bool {
+	hasher := trie.NewStackTrie(nil)
+	computed := types.DeriveSha(types.Receipts(item.Receipts), hasher)
+	return item.Block.ReceiptHash() == computed
+}
+
+type DepositTransaction struct {
+	Sender common.Address
+	To     common.Address
+	Value  uint256.Int
+	Data   []byte
+}
+
+func (dep *DepositTransaction) Decode(data []byte) (valid bool) {
+	// TODO
+	return false
+}
+
+type OpaqueTransaction []byte
+
+type L2Block struct {
```

Comment:
Is this an "L2BlockInput" ? Or is this a stand-in for the actual L2 blocks?

Reply:
If this is a real L2 block, should this just be a normal `Block` type?

Reply:
It's just the input, the contents a sequencer posts on-chain. I'll remove it for now while we're only doing the deposit-rollup. I'm curious though, how do we determine the coinbase address? Do we encode it as part of the transactions batch, or do we maybe use the sender of the transaction that made the data available?

Reply:
Probably better to discuss in an issue later. Will focus on deposit-rollup for now.